packages/svelte/src/internal/client/dom/blocks/each.js

```diff
@@ -204,342 +213,170 @@ function each(anchor, flags, get_collection, get_key, render_fn, fallback_fn, re
-			// destroy everything first, then append new items
-			pause_effects(to_destroy, anchor);
-			to_destroy = [];
-		}
+			matched = [];
+			stashed = [];
 
-		// working from the back, insert new or moved items
-		while (b-- > start) {
-			index = sources[b - start];
-			var should_insert = index === NEW_ITEM;
-
-			if (should_insert) {
-				if (last_item !== undefined) anchor = get_first_child(last_item);
-				item = create_item(anchor, array[b], keys[b], b, render_fn, flags);
-			} else {
-				item = b_items[b];
-				if (should_update) {
-					update_item(item, array[b], b, flags);
-				}
+			while (current && current.k !== key) {
+				seen.add(current);
+				stashed.push(current);
+				current = current.next;
+			}
 
-				if (moved && index !== LIS_ITEM) {
-					if (last_item !== undefined) anchor = get_first_child(last_item);
-					move(/** @type {import('#client').Dom} */ (item.e.dom), anchor);
-				}
+			if (current === null) {
+				continue;
 			}
 
-			last_item = b_items[b] = item;
+			item = current;
 		}
-	}
 
-	if (to_animate.length > 0) {
-		// TODO we need to briefly take any outroing elements out of the flow, so that
-		// we can figure out the eventual destination of the animating elements
-		// - https://github.com/sveltejs/svelte/pull/10798#issuecomment-2013681778
-		// - https://svelte.dev/repl/6e891305e9644a7ca7065fa95c79d2d2?version=4.2.9
-		effect(() => {
-			untrack(() => {
-				for (item of to_animate) {
-					item.a?.apply();
-				}
-			});
-		});
+		matched.push(item);
+		prev = item;
+		current = item.next;
 	}
 
-	var controlled_anchor = is_controlled && b_items.length === 0 ? anchor : null;
-
-	pause_effects(to_destroy, controlled_anchor, () => {
-		state.items = b_items;
-	});
-}
-
-/**
- * Longest Increased Subsequence algorithm
- * @param {Int32Array} a
- * @returns {void}
- */
-function mark_lis(a) {
-	var length = a.length;
-	var parent = new Int32Array(length);
-	var index = new Int32Array(length);
-	var index_length = 0;
-	var i = 0;
-
-	/** @type {number} */
-	var j;
-
-	/** @type {number} */
-	var k;
-
-	/** @type {number} */
-	var lo;
-
-	/** @type {number} */
-	var hi;
-
-	// Skip -1 values at the start of the input array `a`.
-	for (; a[i] === NEW_ITEM; ++i) {
-		/**/
+	while (current) {
+		seen.add(current);
+		current = current.next;
 	}
 
-	index[0] = i++;
-
-	for (; i < length; ++i) {
-		k = a[i];
-
-		if (k !== NEW_ITEM) {
-			// Ignore -1 values.
-			j = index[index_length];
-
-			if (a[j] < k) {
-				parent[i] = j;
-				index[++index_length] = i;
-			} else {
-				lo = 0;
-				hi = index_length;
+	const to_destroy = Array.from(seen);
 
-				while (lo < hi) {
-					j = (lo + hi) >> 1;
-					if (a[index[j]] < k) {
-						lo = j + 1;
-					} else {
-						hi = j;
-					}
-				}
-
-				if (k < a[index[lo]]) {
-					if (lo > 0) {
-						parent[i] = index[lo - 1];
-					}
-					index[lo] = i;
-				}
+	pause_effects(
+		to_destroy.map((item) => item.e),
```

Comment:
I wonder if we can avoid this and just pass in the items instead? Given that `pause_effects` will just loop over them all anyway.